### ISSUE TITLE
Average Measure for Lasso and XGBoost survival models 

### ISSUE THREAD

**Issue by aalhadab** [mlr-org/mlr3proba#233]
I am not sure if this is a bug or a problem with my code. When I tried to calculate the measure for different survival models for train and test set in resample I get NAN for lasso and xgboost. 

set.seed(11)
split = initial_split(tsk("actg")$data(), strata = status )
train = training(split) 
test  = testing (split)

task  = TaskSurv$new(id = "actg", backend = train, time = "time", event = "status")
task$col_roles$stratum  = c("txgrp", "status")

CV 
set.seed(21)
cv = rsmp("cv", folds=5)$instantiate(task)

Evaluation measure 
measure = list(msr("surv.cindex", predict_sets ="train", id="train_C-index"),
               msr("surv.cindex", predict_sets ="test",  id="test_C-index"))

learners 
cox    = lrn("surv.coxph",      predict_sets = c("train", "test"))
aft    = lrn("surv.parametric", predict_sets = c("train", "test"))
lasso  = as_learner(po("encode") %>>% lrn("surv.glmnet", predict_sets = c("train", "test")))
rsf    = lrn("surv.rfsrc", predict_sets = c("train", "test"))
xgboost= as_learner(po("encode") %>>% lrn("surv.xgboost", predict_sets = c("train", "test")))
gbm    = lrn("surv.gbm", predict_sets = c("train", "test"))

fit models
resample(task, cox, cv)$aggregate(measure)
resample(task, aft, cv)$aggregate(measure)
resample(task, lasso, cv)$aggregate(measure)
resample(task, rsf, cv)$aggregate(measure)
resample(task, xgboost, cv)$aggregate(measure)
resample(task, gbm, cv)$aggregate(measure)

**Comment by RaphaelS1** [comment]
Hi @aalhadab can you please provide a [reprex](https://stackoverflow.com/help/minimal-reproducible-example) using [{reprex}](https://cran.r-project.org/web/packages/reprex/index.html) or at least just wrap your example with `reprex::reprex` so I can see your session info properly

**Comment by aalhadab** [comment]
``` r
library(mlr3verse)
#> Warning: package 'mlr3verse' was built under R version 4.0.5
#> Loading required package: mlr3
#> Warning: package 'mlr3' was built under R version 4.0.5
devtools::install_github("mlr-org/mlr3extralearners")
#> Skipping install of 'mlr3extralearners' from a github remote, the SHA1 (6e556d96) has not changed since last install.
#>   Use `force = TRUE` to force installation
library(mlr3extralearners)
#> 
#> Attaching package: 'mlr3extralearners'
#> The following objects are masked from 'package:mlr3':
#> 
#>     lrn, lrns
library(pracma)
#> Warning: package 'pracma' was built under R version 4.0.5
library(tidymodels)
#> Warning: package 'tidymodels' was built under R version 4.0.5
#> Registered S3 method overwritten by 'tune':
#>   method                   from   
#>   required_pkgs.model_spec parsnip
#> Warning: package 'broom' was built under R version 4.0.5
#> Warning: package 'dials' was built under R version 4.0.5
#> Warning: package 'scales' was built under R version 4.0.5
#> Warning: package 'dplyr' was built under R version 4.0.5
#> Warning: package 'ggplot2' was built under R version 4.0.5
#> Warning: package 'infer' was built under R version 4.0.5
#> Warning: package 'modeldata' was built under R version 4.0.5
#> Warning: package 'parsnip' was built under R version 4.0.5
#> Warning: package 'purrr' was built under R version 4.0.5
#> Warning: package 'recipes' was built under R version 4.0.5
#> Warning: package 'rsample' was built under R version 4.0.5
#> Warning: package 'tibble' was built under R version 4.0.5
#> Warning: package 'tidyr' was built under R version 4.0.5
#> Warning: package 'tune' was built under R version 4.0.5
#> Warning: package 'workflows' was built under R version 4.0.5
#> Warning: package 'workflowsets' was built under R version 4.0.5
#> Warning: package 'yardstick' was built under R version 4.0.5
library(tidyverse)
#> Warning: package 'tidyverse' was built under R version 4.0.5
#> Warning: package 'readr' was built under R version 4.0.5
#> Warning: package 'forcats' was built under R version 4.0.5

# Task 
set.seed(11)
split = initial_split(tsk("actg")$data(), strata = status )
train = training(split) 
test  = testing (split)

task  = TaskSurv$new(id = "actg", backend = train, time = "time", event = "status")
task$col_roles$stratum  = c("txgrp", "status")

# CV 
set.seed(21)
cv = mlr3::rsmp("cv", folds=5)$instantiate(task)

# Evaluation measure 
measure = list(
  msr("surv.cindex", predict_sets = "train", id = "train_cindex"),
  msr("surv.cindex", predict_sets = "test",  id = "test_cindex"))

# learners 
cox    = lrn("surv.coxph",  predict_sets = c("train", "test"))
resample(task, cox, cv)$aggregate(measure)
#> INFO  [20:08:47.912] [mlr3] Applying learner 'surv.coxph' on task 'actg' (iter 4/5) 
#> INFO  [20:08:48.069] [mlr3] Applying learner 'surv.coxph' on task 'actg' (iter 5/5) 
#> INFO  [20:08:48.183] [mlr3] Applying learner 'surv.coxph' on task 'actg' (iter 1/5) 
#> INFO  [20:08:48.269] [mlr3] Applying learner 'surv.coxph' on task 'actg' (iter 2/5) 
#> INFO  [20:08:48.372] [mlr3] Applying learner 'surv.coxph' on task 'actg' (iter 3/5)
#> Warning in coxph.fit(X, Y, istrat, offset, init, control, weights = weights, :
#> Loglik converged before variable 4,13,16,17,18 ; coefficient may be infinite.
#> Warning in coxph.fit(X, Y, istrat, offset, init, control, weights = weights, :
#> Loglik converged before variable 4,16,17,18 ; coefficient may be infinite.
#> Warning in coxph.fit(X, Y, istrat, offset, init, control, weights = weights, :
#> Loglik converged before variable 4,16,17 ; coefficient may be infinite.
#> Warning in coxph.fit(X, Y, istrat, offset, init, control, weights = weights, :
#> Loglik converged before variable 4,16,17,18 ; coefficient may be infinite.

#> Warning in coxph.fit(X, Y, istrat, offset, init, control, weights = weights, :
#> Loglik converged before variable 4,16,17,18 ; coefficient may be infinite.
#> train_cindex  test_cindex 
#>    0.7900281    0.7296413


rsf    = lrn("surv.rfsrc", predict_sets = c("train", "test"))
resample(task, rsf, cv)$aggregate(measure)
#> INFO  [20:08:49.073] [mlr3] Applying learner 'surv.rfsrc' on task 'actg' (iter 1/5) 
#> INFO  [20:08:50.306] [mlr3] Applying learner 'surv.rfsrc' on task 'actg' (iter 4/5) 
#> INFO  [20:08:51.528] [mlr3] Applying learner 'surv.rfsrc' on task 'actg' (iter 2/5) 
#> INFO  [20:08:52.682] [mlr3] Applying learner 'surv.rfsrc' on task 'actg' (iter 5/5) 
#> INFO  [20:08:53.888] [mlr3] Applying learner 'surv.rfsrc' on task 'actg' (iter 3/5)
#> train_cindex  test_cindex 
#>    0.9060602    0.7262838


gbm    = lrn("surv.gbm", predict_sets = c("train", "test"))
resample(task, gbm, cv)$aggregate(measure)
#> INFO  [20:08:55.302] [mlr3] Applying learner 'surv.gbm' on task 'actg' (iter 4/5) 
#> INFO  [20:08:55.385] [mlr3] Applying learner 'surv.gbm' on task 'actg' (iter 2/5) 
#> INFO  [20:08:55.440] [mlr3] Applying learner 'surv.gbm' on task 'actg' (iter 1/5) 
#> INFO  [20:08:55.503] [mlr3] Applying learner 'surv.gbm' on task 'actg' (iter 5/5) 
#> INFO  [20:08:55.565] [mlr3] Applying learner 'surv.gbm' on task 'actg' (iter 3/5)
#> Using 100 trees...
#> Using 100 trees...
#> 
#> Using 100 trees...
#> 
#> Using 100 trees...
#> 
#> Using 100 trees...
#> 
#> Using 100 trees...
#> 
#> Using 100 trees...
#> 
#> Using 100 trees...
#> 
#> Using 100 trees...
#> 
#> Using 100 trees...
#> train_cindex  test_cindex 
#>    0.8069223    0.7198504


lasso  = as_learner(po("encode") %>>% lrn("surv.glmnet", predict_sets = c("train", "test")))
resample(task, lasso, cv)$aggregate(measure)
#> INFO  [20:08:56.044] [mlr3] Applying learner 'encode.surv.glmnet' on task 'actg' (iter 2/5) 
#> INFO  [20:08:56.275] [mlr3] Applying learner 'encode.surv.glmnet' on task 'actg' (iter 1/5) 
#> INFO  [20:08:56.443] [mlr3] Applying learner 'encode.surv.glmnet' on task 'actg' (iter 4/5) 
#> INFO  [20:08:56.625] [mlr3] Applying learner 'encode.surv.glmnet' on task 'actg' (iter 5/5) 
#> INFO  [20:08:56.823] [mlr3] Applying learner 'encode.surv.glmnet' on task 'actg' (iter 3/5)
#> Warning: Multiple lambdas have been fit. Lambda will be set to 0.01 (see parameter 's').
#> This happened PipeOp surv.glmnet's $predict()
#> Warning: Multiple lambdas have been fit. Lambda will be set to 0.01 (see parameter 's').
#> This happened PipeOp surv.glmnet's $predict()

#> Warning: Multiple lambdas have been fit. Lambda will be set to 0.01 (see parameter 's').
#> This happened PipeOp surv.glmnet's $predict()

#> Warning: Multiple lambdas have been fit. Lambda will be set to 0.01 (see parameter 's').
#> This happened PipeOp surv.glmnet's $predict()

#> Warning: Multiple lambdas have been fit. Lambda will be set to 0.01 (see parameter 's').
#> This happened PipeOp surv.glmnet's $predict()
#> Warning: Measure 'train_cindex' needs predict sets 'train', but learner
#> 'encode.surv.glmnet' only predicted on sets 'test'
#> train_cindex  test_cindex 
#>          NaN     0.748525


xgboost= as_learner(po("encode") %>>% lrn("surv.xgboost", predict_sets = c("train", "test")))
resample(task, xgboost, cv)$aggregate(measure)
#> INFO  [20:08:57.547] [mlr3] Applying learner 'encode.surv.xgboost' on task 'actg' (iter 5/5) 
#> INFO  [20:08:57.710] [mlr3] Applying learner 'encode.surv.xgboost' on task 'actg' (iter 2/5) 
#> INFO  [20:08:57.858] [mlr3] Applying learner 'encode.surv.xgboost' on task 'actg' (iter 1/5) 
#> INFO  [20:08:57.997] [mlr3] Applying learner 'encode.surv.xgboost' on task 'actg' (iter 4/5) 
#> INFO  [20:08:58.129] [mlr3] Applying learner 'encode.surv.xgboost' on task 'actg' (iter 3/5)
#> Warning: Measure 'train_cindex' needs predict sets 'train', but learner
#> 'encode.surv.xgboost' only predicted on sets 'test'
#> train_cindex  test_cindex 
#>          NaN    0.6551361
```

<sup>Created on 2021-12-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by RaphaelS1** [comment]
Not seeing the problem you have your test measure and the warning shows the prediction only works on test set only? This is neither a bug nor a problem with your code (training error isn't particularly informative anyway). I will close this issue unless you have further questions about this
